src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -481,7 +481,35 @@ async Task<DurableOrchestrationStatus> IDurableOrchestrationClient.GetStatusAsyn
                 return null;
             }
 
-            return await GetDurableOrchestrationStatusAsync(state, this.client, showHistory, showHistoryOutput);
+            return await GetDurableOrchestrationStatusAsync(state, this.client, showHistory, showHistoryOutput, showHistoryInput);
+        }
+
+        /// <inheritdoc/>
+        async Task<DurableOrchestrationStatus> IDurableOrchestrationClient.QueryStatusAsync(string instanceId, InstanceFilterOption instanceFilterOption)
```

Comment:
Let's not introduce a new method. I think we can accomplish our goals as I mentioned earlier more cheaply if we simply rely on the broader interpretation of the existing `showInput` parameter.